### ISSUE TITLE
Expanded answer structures for SingleListGrader

### ISSUE THREAD

**Issue by jolyonb** [mitodl/mitx-grading-library#247]
This follows from #3.

In order to give feedback on lists, I'm going to propose allowing for a dictionary to describe the answer for a list in the following way:
```python
answers = {'expect': [list of answers here], 'grade_decimal': 0.5, 'msg': 'Overall message here'}
```
The message would display only if all answers are correct; the `grade_decimal` value would multiply every box. One could also have tuples of such dictionaries. Just giving a list would be equivalent to having the expect keyword.

Note that this doesn't work nearly so nicely for ListGrader due to nested lists not having a nice place to put overall messages for sublists.

**Comment by jolyonb** [comment]
Related to this is for a singlelistgrader to read its answer from the expect keyword.

**Comment by ChristopherChudzicki** [comment]
Resolved, right?

**Comment by jolyonb** [comment]
Should be all resolved :-)

**Comment by ChristopherChudzicki** [comment]
Resolved by #261
